### PR TITLE
Swift: remove unneeded properties from `InterpolatedStringLiteralExpr`

### PR DESCRIPTION
**Pull request by redsun82**
These properties were unused in the QL library (hence the full upgrade/downgrade compatibility).

### CODE REVIEW

**Review by MathiasVP — DISMISSED** [review]
LGTM!

File: swift/ql/lib/upgrades/ba4171b90d0665b40e9e203bac9e3d4a0b2d03ec/integer_literal_exprs.ql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+from Element i, string value
+where
+  integer_literal_exprs(i, value) and
+  not exists(Element interpolated |
+    interpolated_string_literal_expr_interpolation_count_exprs(interpolated, i)
+    or
+    interpolated_string_literal_expr_literal_capacity_exprs(interpolated, i)
+  )
+select i, value
```
Comment (geoffw0) [inline swift/ql/lib/upgrades/ba4171b90d0665b40e9e203bac9e3d4a0b2d03ec/integer_literal_exprs.ql:13]: So if I'm understanding correctly, the `interpolated_string_literal_expr_interpolation_count_exprs` and `interpolated_string_literal_expr_literal_capacity_exprs` didn't actually have numerical (`int`) values, but linked to expressions that were (always?) an integer literal holding that value?

File: swift/downgrades/44e36e15e90bc1535964d9b86b3cd06a8b0d26e3/upgrade.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+description: Revert removing `getInterpolationCountExpr` and `getLiteralCapacityExpr` from `InterpolatedStringLiteralExpr`
+compatibility: full
```
Comment (geoffw0) [inline swift/downgrades/44e36e15e90bc1535964d9b86b3cd06a8b0d26e3/upgrade.properties:2]: So the downgrade is to do nothing, giving us an empty `interpolated_string_literal_expr_interpolation_count_exprs` and `interpolated_string_literal_expr_literal_capacity_exprs` (and no associated integer literal expressions).  I see that we don't actually use these in queries, but they were exposed (e.g. by `getInterpolationCountExpr()`), so I think perhaps `backwards` compatibility might be more accurate for the downgrade script???

**Review by geoffw0 — COMMENTED** [review]
A couple of questions, but LGTM.

File: swift/downgrades/44e36e15e90bc1535964d9b86b3cd06a8b0d26e3/upgrade.properties (hunk 2 above)
Comment (geoffw0) [reply swift/downgrades/44e36e15e90bc1535964d9b86b3cd06a8b0d26e3/upgrade.properties:2]: 100% agree this is not super important, and shouldn't hold up merging the PR.

File: swift/ql/lib/upgrades/ba4171b90d0665b40e9e203bac9e3d4a0b2d03ec/integer_literal_exprs.ql (hunk 1 above)
Comment (redsun82) [reply swift/ql/lib/upgrades/ba4171b90d0665b40e9e203bac9e3d4a0b2d03ec/integer_literal_exprs.ql:13]: exactly, they were always integer literals, as documented in the Swift compiler here:

https://github.com/apple/swift/blob/f5361ed68f32163d580dabf7bb596ea39c6b27a5/include/swift/AST/Expr.h#L938C1-L946

File: swift/downgrades/44e36e15e90bc1535964d9b86b3cd06a8b0d26e3/upgrade.properties (hunk 2 above)
Comment (redsun82) [reply swift/downgrades/44e36e15e90bc1535964d9b86b3cd06a8b0d26e3/upgrade.properties]: changed to backwards :+1:

**Comment by redsun82** [comment]
DCA looks good as far as I'm concerned, this can be merged as soon as CI is happy.

**Review by geoffw0 — APPROVED** [review]
LGTM.

**Pull request merged**